src/hipscat/pixel_math/box_filter.py

```diff
@@ -0,0 +1,146 @@
+from __future__ import annotations
+
+from typing import List, Tuple
+
+import astropy.units as u
+import healpy as hp
+import numpy as np
+from astropy.coordinates import Angle
+
+from hipscat.pixel_math import HealpixPixel
+from hipscat.pixel_math.filter import get_filtered_pixel_list
+from hipscat.pixel_math.polygon_filter import SphericalCoordinates
+from hipscat.pixel_tree.pixel_tree import PixelTree
+from hipscat.pixel_tree.pixel_tree_builder import PixelTreeBuilder
+
+
+def filter_pixels_by_box(
+    pixel_tree: PixelTree, ra: Tuple[float, float] | None, dec: Tuple[float, float] | None
+) -> List[HealpixPixel]:
+    """Filter the leaf pixels in a pixel tree to return a partition_info dataframe
+    with the pixels that overlap with a right ascension or the declination region.
+    Only one of ra and dec should be set, otherwise one should use polygonal search.
+
+    Args:
+        pixel_tree (PixelTree): The catalog tree to filter pixels from
+        ra (Tuple[float, float]): Right ascension range, in [0,360] degrees
+        dec (Tuple[float, float]): Declination range, in [-90,90] degrees
+
+    Returns:
+        List of HealpixPixels representing only the pixels that overlap with the right
+        ascension or the declination region.
+    """
+    max_order = pixel_tree.get_max_depth()
+
+    result_pixels = []
+    ra_search_tree, dec_search_tree = None, None
+
+    if ra is not None:
+        ra_search_tree = _generate_ra_strip_pixel_tree(ra, max_order)
+        result_pixels = get_filtered_pixel_list(pixel_tree, ra_search_tree)
+    if dec is not None:
+        dec_search_tree = _generate_dec_strip_pixel_tree(dec, max_order)
+        result_pixels = get_filtered_pixel_list(pixel_tree, dec_search_tree)
+    if ra_search_tree is not None and dec_search_tree is not None:
+        filter_pixels = get_filtered_pixel_list(ra_search_tree, dec_search_tree)
+        filter_tree = PixelTreeBuilder.from_healpix(filter_pixels)
+        result_pixels = get_filtered_pixel_list(pixel_tree, filter_tree)
+
+    return result_pixels
+
+
+def wrap_ra_values(ra: Tuple[float, float] | None) -> Tuple[float, float] | None:
+    """Wraps right ascension values to the [0,360] degree range and sorts
+    declination values by ascending order.
```

Comment:
Update comment: This doesn't change any dec values.

Reply:
Same here, thanks for noticing!